packages/now-node/src/helpers.ts

```diff
@@ -72,6 +73,46 @@ function getCookieParser(req: NowRequest) {
   };
 }
 
+function parseQSArray(key: string, value: any, acc: any): void {
+  const bktNotation = /\[[^[\]\d]?]/.test(key);
+  const result = /\[(\d*)\]$/.exec(key);
```

Comment:
Why is the regex test() different than the regex for exec()?

Reply:
The test is determining whether the body has bracket notated or index notated arrays. The exec captures either of those cases and outputs the key name (stripping out `[]` or `[0]`)

Reply:
If this can be achieved without regex, I think that would be best in order to avoid [redos](https://en.wikipedia.org/wiki/ReDoS).